plugins/auth-backend-module-bitbucket-provider/src/module.test.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { mockServices, startTestBackend } from '@backstage/backend-test-utils';
+import { authPlugin } from '@backstage/plugin-auth-backend';
+import { authModuleBitbucketProvider } from './module';
+import request from 'supertest';
+import { decodeOAuthState } from '@backstage/plugin-auth-node';
+
+describe('authModuleBitbucketProvider', () => {
+  it('should start', async () => {
+    const { server } = await startTestBackend({
+      features: [
+        authPlugin,
+        authModuleBitbucketProvider,
+        mockServices.rootConfig.factory({
+          data: {
+            app: {
+              baseUrl: 'http://localhost:3000',
+            },
+            auth: {
+              providers: {
+                bitbucket: {
+                  development: {
+                    clientId: 'my-client-id',
+                    clientSecret: 'my-client-secret',
+                  },
+                },
+              },
+            },
+          },
+        }),
+      ],
+    });
+
+    const agent = request.agent(server);
+
+    const res = await agent.get('/api/auth/bitbucket/start?env=development');
+
+    expect(res.status).toEqual(302);
+
+    const nonceCookie = agent.jar.getCookie('bitbucket-nonce', {
+      domain: 'localhost',
+      path: '/api/auth/bitbucket/handler',
+      script: false,
+      secure: false,
+    });
+    expect(nonceCookie).toBeDefined();
+
+    const startUrl = new URL(res.get('location'));
+    expect(startUrl.origin).toBe('https://bitbucket.org');
+    expect(startUrl.pathname).toBe('/oauth/authorize');
+    expect(Object.fromEntries(startUrl.searchParams)).toEqual({
```

Comment:
@Rugvip this part of the test keeps failing after my PR goes through the GitHub action checks. Could the issue perhaps be from the pathname `toBe('/oath/authorize)`

Reply:
@Rugvip

Reply:
Looks like there's an extra `scope` param being checked for down on line 68 that doesn't exist. I'd assume that's a remnant from the GitLab provider, which adds that scope unconditionally within the passport strategy? Does it work if you simply remove line 68?

Reply:
Yes! Thank you.